### PR TITLE
Support exchanging custom types.

### PR DESCRIPTION
**Pull request by cescoffier**
Fix issue #11

### CODE REVIEW

**Comment by cescoffier** [comment]
@davsclaus can you have a quick look.

File: src/main/java/io/vertx/ext/camel/impl/CamelToVertxProcessor.java [hunk 1]
```diff
@@ -54,30 +54,47 @@ public boolean process(Exchange exchange, AsyncCallback callback) {
...
+
     DeliveryOptions delivery = CamelHelper.getDeliveryOptions(in, inbound.isHeadersCopy());
 
     try {
       if (inbound.isPublish()) {
-        vertx.eventBus().publish(inbound.getAddress(), body.toString(), delivery);
+        try {
+          vertx.eventBus().publish(inbound.getAddress(), body, delivery);
+        } catch (Exception e) {
+          // Mainly there to catch the missing codec exception
```
Comment (davsclaus) [inline src/main/java/io/vertx/ext/camel/impl/CamelToVertxProcessor.java]: If the missing codec exception is thrown by vertx before publishing the message then you should not catch it here, but let it catch at the end of all this block of code, so you return true from the method. As there is no async callback event from vert.x later that would trigger the continued routing on Camel.

File: src/main/java/io/vertx/ext/camel/impl/CamelToVertxProcessor.java [hunk 2]
```diff
@@ -54,30 +54,47 @@ public boolean process(Exchange exchange, AsyncCallback callback) {
...
+                out.setBody(reply.result().body());
+                out.setHeaders(MultiMapHelper.toMap(reply.result().headers()));
+              } else {
+                exchange.setException(reply.cause());
+              }
+              // continue callback
+              callback.done(false);
+            });
+          } catch (Exception e) {
+            // Mainly there to catch the missing codec exception
```
Comment (davsclaus) [inline src/main/java/io/vertx/ext/camel/impl/CamelToVertxProcessor.java]: Same problem as mentioned in previous comment

Comment (cescoffier) [inline src/main/java/io/vertx/ext/camel/impl/CamelToVertxProcessor.java]: Yes, my bad, did'n see the global exception handler. It's fixed.

**Comment by davsclaus** [comment]
Looks good

**Pull request merged**